Updating Expo/Firestore docs

masaok opened the pull request:
# Why

Expo supports Firestore, but the latest docs have not been updated.  After chatting with Brent and Evan in Slack, I decided to contribute to the docs.

# How

Built using Expo on Mac (iOS) and Windows (Android)

# Test Plan

Hmmm... no test plan yet.  Although it works in the Expo app that I'm working on.  If you have any suggestions, please let me know!

masaok committed:
Specify that the example is for the Realtime Database

(not Firestore)

masaok committed:
adding Firestore data update example

masaok committed:
adding a period

masaok committed:
updating the "note"

masaok committed:
improving the sample code

masaok committed:
Merge branch 'master' of https://github.com/expo/expo into patch-1

EvanBacon commented:
thanks for this

EvanBacon commented on docs/pages/versions/v32.0.0/guides/using-firebase.md:
```diff
@@ -8,7 +8,7 @@ Luckily, the Firebase JavaScript SDK starting from version 3.1+ has almost full
 See [firebase.google.com/docs/database](https://firebase.google.com/docs/database) for more general information and the [official Firebase blog post announcing React Native compatibility](https://firebase.googleblog.com/2016/07/firebase-react-native.html)
 
-> **Note:** This guide only covers Firebase Realtime Database, and not the other services under the larger Google Firebase umbrella. Firebase Cloud Storage is currently not supported, but we are [working on upstreaming a Blob implementation](https://github.com/facebook/react-native/issues/11103) to React Native that would make this possible. For more background on why other Firebase services are  ...
+> **Note:** This guide mostly covers Firebase Realtime Database (and some Firestore as well), but not the other services under the larger Google Firebase umbrella. Firebase Cloud Storage is currently not supported, but we are [working on upstreaming a Blob implementation](https://github.com/facebook/react-native/issues/11103) to React Native that would make this possible. For more background on w ...
```
could you update this, storage has been possible for some time now

masaok replied:
Sure, I will try, although I've never used it myself (I've only seriously been using Expo for a few weeks now).  Looks like this is the package: https://www.npmjs.com/package/expo-firebase-storage ... yes?

EvanBacon replied:
nope you can use web storage with Expo. That library is used for native firebase storage.

EvanBacon replied:
https://github.com/EvanBacon/firebase-instagram/blob/master/utils/uploadPhoto.js

masaok replied:
As I stated in my commit, I added text that I thought would be helpful to future "new people" (myself included) who are just starting to use Expo with Firebase (and might be having trouble navigating the docs and code).

EvanBacon commented on docs/pages/versions/v32.0.0/guides/using-firebase.md:
```diff
@@ -166,3 +166,26 @@ function storeHighScore(user, score) {
+dbh.collection("characters").doc("mario").set({
+  employment: "plumber",
+  outfitColor: "red",
+  specialAttack: "fireball"
```
nice

masaok replied:
thanks!

EvanBacon reviewed (changes requested):
Just update the note about storage not being available.

masaok committed:
Fixed the note, added more links

(this is from a "new guy" perspective, who just realized where to find the "official list" of all expo-* packages)

masaok committed:
sample code is *not* an "authenticated update"

I had just copied the subheading from the previous subheading.  Now, it's more accurate.

masaok committed:
fixing intro paragraphs

...since the last update was two years ago.

masaok committed:
testing table of contents

masaok committed:
table of contents works!

masaok committed:
rephrasing and re-ordering Firestore text

EvanBacon commented:
Looks good. Could you move the changes to this file: `docs/pages/versions/unversioned/guides/using-firebase.md` Sorry, I didn't notice this sooner.

masaok commented:
> Looks good. Could you move the changes to this file: `docs/pages/versions/unversioned/guides/using-firebase.md` Sorry, I didn't notice this sooner.

Move or copy?  I'll copy first, then revert the v32 file if you confirm you prefer a "move".

masaok committed:
Merge branch 'master' of https://github.com/expo/expo into patch-1

masaok committed:
copy doc/guide changes from v32 to unversioned

masaok commented:
> Just update the note about storage not being available.

masaok commented:
@EvanBacon this PR is failing checks at some point in the `publish.sh` script.  I have no idea how the CI stuff works, do you?

![image](https://user-images.githubusercontent.com/1320083/55062173-a5b1f180-5032-11e9-88f0-de1dbc6d82b3.png)

brentvatne commented:
the failure should be unrelated to docs, thanks!
